services/device/geolocation/geoclue_location_provider.cc

```diff
@@ -0,0 +1,414 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/services/device/geolocation/geoclue_location_provider.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/barrier_callback.h"
+#include "base/feature_list.h"
+#include "base/functional/bind.h"
+#include "base/functional/callback_forward.h"
+#include "base/functional/callback_helpers.h"
+#include "base/location.h"
+#include "base/memory/weak_ptr.h"
+#include "base/ranges/algorithm.h"
+#include "base/time/time.h"
+#include "components/dbus/thread_linux/dbus_thread_linux.h"
+#include "dbus/property.h"
+#include "services/device/public/cpp/device_features.h"
+#include "services/device/public/cpp/geolocation/geolocation_manager.h"
+#include "services/device/public/cpp/geolocation/geoposition.h"
+
+namespace device {
+
+namespace {
+
+constexpr char kServiceName[] = "org.freedesktop.GeoClue2";
+constexpr char kLocationInterfaceName[] = "org.freedesktop.GeoClue2.Location";
+constexpr char kClientInterfaceName[] = "org.freedesktop.GeoClue2.Client";
+constexpr char kManagerInterfaceName[] = "org.freedesktop.GeoClue2.Manager";
+constexpr char kManagerObjectPath[] = "/org/freedesktop/GeoClue2/Manager";
+constexpr char kBraveDesktopId[] = "com.brave.Browser";
+
+// This corresponds to the GeoClue2 AccuracyLevel: Neighborhood
+constexpr uint32_t kGeoClueAccuracyLow = 5;
+
+// This corresponds to the GeoClue2 AccuracyLevel: Exact
+constexpr uint32_t kGeoClueAccuracyHigh = 8;
+
+mojom::Geoposition GetErrorPosition() {
```

Comment:
these methods should be inside an anonymous`namespace { .. }`

Reply:
I mean everything that is not declared in the header file including `GetErrorPosition` and `GeoClueAvailable`

Reply:
Okay, sweet, I've moved `GeoClueProperties` into the anonymouse namespace.